app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -24,6 +24,8 @@ def complete
     raise Discourse::NotFound unless request.env["omniauth.auth"]
 
     auth[:session] = session
+    auth[:client_ip] = request.remote_ip
+    auth[:user_agent] = request.user_agent
```

Comment:
We already pass the session object here, so maybe we could just pass the whole `request` object as well? That way, plugins are free to look at whatever information they need, with minimal code in core:

```suggestion
    auth[:session] = session
    auth[:request] = request
```